src/main/java/org/folio/innreach/domain/service/impl/InnReachTransactionActionServiceImpl.java

```diff
@@ -129,30 +134,46 @@ public void associateNewLoanWithTransaction(LoanDTO loan) {
 
   @Override
   public void handleLoanUpdate(LoanDTO loan) {
+    InnReachTransaction associatedTransactionWithLoan = null;
```

Comment:
rename it to `transaction`, it should be obvious from the context that the transaction is associated with loan

Reply:
use `var` consistently

Reply:
var doesn't work here

Reply:
```
    var transaction = transactionRepository.fetchOneByLoanId(loan.getId()).orElse(null);
    if (transaction == null) {
      return;
    }
```

Reply:
I am going to use  if (optionalTransaction.isPresent()) {
      transaction = optionalTransaction.get();
    } else {
      new EntityNotFoundException("INN-Reach transaction is not found by loan id: " + loan.getId());
    } isn't ok ?

Reply:
Got it